src/timer.rs

```diff
@@ -147,424 +124,540 @@ pub enum Event {
-                #[inline]
-                pub fn enable_interrupt(&mut self, event: Event) {
-                    self.configure_interrupt(event, true);
-                }
+    /// Clear the given interrupt event flag.
+    #[inline]
+    pub fn clear_event(&mut self, event: Event) {
+        match event {
+            Event::Update => self.tim.sr.modify(|_, w| w.uif().clear()),
+        }
+    }
 
-                /// Enable or disable the interrupt for the specified [`Event`].
-                #[inline]
-                pub fn disable_interrupt(&mut self, event: Event) {
-                    self.configure_interrupt(event, false);
-                }
+    /// Clear **all** interrupt events.
+    #[inline]
+    pub fn clear_events(&mut self) {
+        // SAFETY: This atomic write clears all flags and ignores the reserverd bit fields.
+        self.tim.sr.write(|w| unsafe { w.bits(0) });
+    }
 
+    /// Releases the TIM peripheral
+    #[inline]
+    pub fn free(mut self) -> TIM {
+        self.stop();
+        self.tim
+    }
+}
 
-                // TODO(Sh3Rm4n): After macro -> Instance refactor introduce associated const INTERRUPT.
-                // /// Get the correspoing Interrupt number
-                // // pub fn nvic() -> Interrupt { }
+impl<TIM> Periodic for Timer<TIM> where TIM: Instance {}
 
-                /// Enable or disable the interrupt for the specified [`Event`].
-                #[inline]
-                pub fn configure_interrupt(&mut self, event: Event, enable: bool) {
-                    match event {
-                        Event::Update => self.tim.dier.write(|w| w.uie().bit(enable)),
-                    }
-                }
+impl<TIM> CountDown for Timer<TIM>
+where
+    TIM: Instance,
+{
+    type Time = Hertz;
 
-                /// Enable or disable interrupt for the specified [`Event`]s.
-                ///
-                /// Like [`Timer::configure_interrupt`], but instead using an enumset. The corresponding
-                /// interrupt for every [`Event`] in the set will be enabled, every other interrupt will be
-                /// **disabled**.
-                #[cfg(feature = "enumset")]
-                #[cfg_attr(docsrs, doc(cfg(feature = "enumset")))]
-                pub fn configure_interrupts(&mut self, events: EnumSet<Event>) -> &mut Self {
-                    for event in events.complement().iter() {
-                        self.configure_interrupt(event, false);
-                    }
-                    for event in events.iter() {
-                        self.configure_interrupt(event, true);
-                    }
+    fn start<T>(&mut self, timeout: T)
+    where
+        T: Into<Self::Time>,
+    {
+        self.stop();
 
-                    self
-                }
+        let frequency = timeout.into().integer();
+        let timer_clock = TIM::clock(&self.clocks);
+        // TODO: Obtain ppre1 and pclk1 (is ppre1 even the correct ppre???)
```

Comment:
> is ppre1 even the correct ppre???

TIM peripherals on APB1 uses ppre1 and on APB2 uses ppre2

Reply:
Thanks for the quick lookup. I suspected something like this.